tmx_wangset.go

```diff
@@ -0,0 +1,109 @@
+package tiled
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// WangSets contains the list of Wang sets defined for this tileset.
+// https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#wangsets
+// Can contain any number: <wangset>
+type WangSets []*WangSet
+
+// WangSet defines a list of corner colors and a list of edge colors, and any number of Wang tiles using these colors.
+type WangSet struct {
+	Name       string       `xml:"name,attr"` // The name of the Wang set.
+	Type       string       `xml:"type,attr"` // ex. corner
+	TileId     uint32       `xml:"tile,attr"` // The tile ID of the tile representing this Wang set.
+	WangColors []*WangColor `xml:"wangcolor"`
+	WangTiles  []*WangTile  `xml:"wangtile"`
+}
+
+// Wangcolor that can be used to define the corner and/or edge of a Wang tile.
+type WangColor struct {
+	Name        string  `xml:"name,attr"`        //  The name of this color.
+	Color       string  `xml:"color,attr"`       // The color in #RRGGBB format (example: #c17d11).
+	TileID      uint32  `xml:"tile,attr"`        // The tile ID of the tile representing this color.
+	Probability float32 `xml:"probability,attr"` // The relative probability that this color is chosen over others in case of multiple options. (defaults to 0)
+}
+
+// Wangtile, by referring to a tile in the tileset and associating it with a certain Wang ID.
+type WangTile struct {
+	TileID uint32 `xml:"tileid,attr"` // The tile ID.
+
+	// WangID, given by a comma-separated list of indexes (starting from 1, because 0 means _unset_)
+	// referring to the Wang colors in the Wang set in the following order:
+	// top, top right, right, bottom right, bottom, bottom left, left, top left (since Tiled 1.5).
+	// Before Tiled 1.5, the Wang ID was saved as a 32-bit unsigned integer stored in the format
+	// 0xCECECECE (where each C is a corner color and each E is an edge color, in reverse order).
+	WangID string `xml:"wangid,attr"` //
+}
+
+// WangPosition Wang Color mapping to position
+type WangPosition int
+
+const (
+	Top WangPosition = iota
```

Comment:
It should be iota+1 as it should start with 1 to be in par with spec

Reply:
The spec specifies that a WangID of 0 means _unset_. So for example if the wangTile is of corner type the WangID string would be "0,1,0,3,0,1,0,1"  where 0 just means there is nothing set for the borders only the corners.
Each number means what type of WangColor for each position. in this order:
top, top right, right, bottom right, bottom, bottom left, left, top left 

I wrote the GetWangColors() because that string is quite confusing to look at. So instead of just returning a CSV it now returns a map where the keys are the position and the values are the WangColor.

So its the value that is potentially _unset_ not the key. 
So the enum can start at 0 since it just reflects the 0 index in the position list but if a WangColor is _unset_ we return nil for that position.

Reply:
The numbers in "0,1,0,3,0,1,0,1" just means that the top right, bottom left and top left are the same wangColor of id 1.
and Bottom right is a different WangColor of id 3

Reply:
Its confusing and I wish Tiled just exported the Terrain types instead but I don't know how to get it to do that